src/ddmd/semantic.d

```diff
@@ -0,0 +1,50 @@
+module ddmd.semantic;
+
+import ddmd.dsymbol;
+import ddmd.dsymbolsem;
+import ddmd.dscope;
+import ddmd.expression;
+import ddmd.expressionsem;
+import ddmd.init;
+import ddmd.initsem;
+import ddmd.mtype;
+import ddmd.statement;
+import ddmd.statementsem;
+
+/*************************************
+ * Does semantic analysis on the public face of declarations.
+ */
+extern(C++) void semantic(Dsymbol dsym, Scope* sc)
```

Comment:
This looks like a new function. Please use common Ddoc sections like summary, description and parameters.